File: poo/Cpp/MaiorPalavra/MaiorPalavra.cpp
#include <iostream>
#include <string>

int main() {
    std::string palavra;
    std::string maiorPalavra;
    bool primeira = true;

    while (std::cin >> palavra) {
        if (primeira || palavra.length() > maiorPalavra.length()) {
            maiorPalavra = palavra;
            primeira = false;
        }
    }

    if (primeira) {
        std::cout << "Nenhuma palavra foi informada" << std::endl;
    } else {
        std::cout << "A maior palavra informada foi " << maiorPalavra << std::endl;
    }

    return 0;
}

File: poo/Java/MaiorPalavra/MaiorPalavra.java
import java.util.Scanner;

public class MaiorPalavra {
    public static void main(String[] args) {
        Scanner scanner = new Scanner(System.in);
        String maiorPalavra = "";
        boolean primeira = true;

        while (scanner.hasNext()) {
            String palavra = scanner.next();
            if (primeira) {
                maiorPalavra = palavra;
                primeira = false;
            } else {
                if (palavra.length() > maiorPalavra.length()) {
                    maiorPalavra = palavra;
                }
            }
        }

        if (primeira) {
            System.out.println("Nenhuma palavra foi informada");
        } else {
            System.out.println("A maior palavra informada foi " + maiorPalavra);
        }
        scanner.close();
    }
}